vllm/forward_context.py

```diff
@@ -57,12 +61,37 @@ def set_forward_context(attn_metadata: Any,
     need_to_track_batchsize = track_batchsize and attn_metadata is not None
     if need_to_track_batchsize:
         forward_start_time = time.perf_counter()
+    num_tokens_across_dp = None
+    if vllm_config.parallel_config.data_parallel_size > 1:
+        if attn_metadata is not None:
+            if hasattr(attn_metadata, "num_prefill_tokens"):
+                # for v0 attention backends
+                batchsize = attn_metadata.num_prefill_tokens + \
+                    attn_metadata.num_decode_tokens
+            else:
+                # for v1 attention backends
+                batchsize = attn_metadata.num_input_tokens
+        else:
+            batchsize = num_tokens
+        num_tokens_across_dp = [
+            0 for _ in range(vllm_config.parallel_config.data_parallel_size)
+        ]
```

Comment:
```suggestion
        num_tokens_across_dp = [0] * vllm_config.parallel_config.data_parallel_size
```

Reply:
fixed in [445b474](https://github.com/vllm-project/vllm/pull/13591/commits/445b474161d3459a8bebca9f8957e638bd8f8af5)